agent/action/drain.go

```diff
@@ -81,9 +83,18 @@ func (a DrainAction) Run(drainType DrainType, newSpecs ...boshas.V1ApplySpec) (i
 		scripts = append(scripts, script)
 	}
 
-	parallelScript := a.jobScriptProvider.NewParallelScript("drain", scripts)
-
-	return 0, parallelScript.Run()
+	script := a.jobScriptProvider.NewParallelScript("drain", scripts)
+
+	resultsChan := make(chan error)
```

Comment:
let's make this guy buffered of size 1 so that goroutine can be released if cancelling is done before script finishes running. 

stylistic tip: use 'Ch' suffix for channels.